src/openforms/emails/validators.py

```diff
@@ -28,10 +38,13 @@ def __call__(self, value):
         allowlist = (
             get_system_netloc_allowlist() + config.email_template_netloc_allowlist
         )
+        split_allow_list = {
+            tuple(allowed_domain.split(".")) for allowed_domain in allowlist
+        }
 
         for m in URL_REGEX.finditer(value):
             parsed = urlsplit(m.group())
-            if parsed.netloc not in allowlist:
+            if not is_in_allowlist(split_allow_list, parsed.netloc):
```

Comment:
The assertion is this:
The union of the set of stems of domains of netloc and the set of domains in allowlist must not be empty.
Which means that there exists a stem of netloc that is in allowlist.

the stems for netloc 'open-forms.test.maykin.opengem.nl'
```
{'open-forms.test.maykin.opengem.nl', 'nl', 'opengem.nl', 'maykin.opengem.nl', 'test.maykin.opengem.nl'}
```
Expressing it as tuples of these strings exploded, shows it's not just any string suffix, but only suffices of full tokens (where tokens are domain)....

Maybe formulating it as
```
def subdomains(netloc: str) -> Iterable[str]:
   yield stems from described set of stems

any(stem for stem in subdomains(parsed.netloc) if stem in allowlist)

Reply:
yeah I think I'd prefer Chris' approach to make it easier to read that code, since the tuple splitting is breaking my head + the generator + any approach also does the least amount of work by shortcutting.

Good point about the prefix that may be present in a stem, I didn't think of that.